Update deprecated ansi option to current argument

everte opened the pull request:
Previously docker-compose accepted `--no-ansi`, current versions of docker use `--ansi ("never"|"always"|"auto")`.

The `--no-ansi` causes deprecation warnings.

PS: Sorry for the previous PR, I accidentality committed something extra/unnecessary.

everte committed:
Update deprecated ansi option to current argument

Coffei commented:
@everte Hi and thanks for the contribution!

The diff doesn't look right, you probably wanted to replace the line 159 and not add another line. While i generally agree with replacing `--no-ansi` with newer `--ansi never` we must be careful here, because `--ansi never` is a newer option and might not work with all versions of `docker-compose`. `--no-ansi`, while being deprecated, currently does support more versions and works even with the latest `1.*` version (although it produces a warning).

If you could make the investigation which versions would not be supported with this change that would be great. If not I'll check it out when I have some time. We use this project internally at Altworx, so first I'd have to check if this wouldn't break anything.

Thanks again!

everte commented:
Thanks for your reply.

I'm sorry for the bad diff, that should not have happened (I probably made a small error when selecting the parts of the code I wanted to put in the PR).

I will fix the PR and also investigate which versions of compose do support the newer `--ansi never` option.

I have one unrelated question: I've been adding some extra commands and options that I need (pull, ps, logs, ...). Are you open for pull requests for those changes too?

Coffei commented:
OK, thanks for doing the investigation.

We are very much open for PRs adding new commands. We only implemented what we needed so far, but it makes sense to support more commands. So feel free to open other PRs. :slightly_smiling_face: Thanks!

everte commented:
I've fixed the commit/PR.

Docker-compose added the `--ansi` option in version [1.28 (2021-01-20)](https://docs.docker.com/compose/release-notes/#1282).

I will clean up my code for the other commands I've added and open PR's later.

Coffei commented:
Looks good to me. Thanks for finding the exact version where the option was added, I'll check which versions we use internally to see if this change could pose a problem for us and let you know in the upcoming days. If all will be well we can merge this. Thanks!

Coffei commented:
Hi,
upon closer examination I am not sure we'd want to merge this. We use this library in a docker with official [elixir images](https://hub.docker.com/_/elixir), which are based on [debian bullseye](https://hub.docker.com/_/debian). And the debian has only [docker-compose](https://packages.debian.org/bullseye/docker-compose) 1.25 in its repos. Putting our usecase aside for now, I don't think it's a good reason to drop support for docker-compose available in the official Elixir docker image or stable debian.

Is that fine by you? Is the current implementation with `--no-ansi` causing any issues?

everte commented:
That sounds very reasonable: it is definitely a solid use-case to use docker-compose version from debian bullseye. I apologise for not thinking about this nor checking it and somewhat wasting some of your time.

It was not causing any issues, just clogging my iex logs with deprecated warnings.

Obviously this can be closed.

Coffei commented:
No worries, thank you very much for your interest in the library :) Feel free to open more PRs as you mentioned.
